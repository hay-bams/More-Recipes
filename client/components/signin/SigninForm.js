import React from 'react';
import { Link } from 'react-router-dom';
/**
 * @class Signin
 */
class SigninForm extends React.Component {
    render() {
        return (
            <div className = "container-fluid main-login-container">
            <div className = "row main-login">
              <div className = "col-12 col-sm-8 col-md-5 mx-auto">
                <div className = "card  mt-5">
                  <div className = "card-header">
                    Login
                  </div>
      
                  <div className = "card-body ">
                    <form>
                      <div className = "form-group">
                        <label htmlFor = "email">Email</label>
                        <input type="email" name="email" className = "form-control" placeholder="Enter Email" />
                      </div>
      
                      <div className = "form-group">
                        <label htmlFor = "password">Password</label>
                        <input type="password" name="password" className = "form-control" placeholder="Enter Password" />
                      </div>
      
                      <div className = "form-group">
                        <input type="checkbox" name="checkbox" />
                         Remember me
                      </div>
      
                      <div className = "form-group">
                        <input type="submit" value = "Log in" className = "btn btn-success btn-block" />
                      </div>
      
                       <div className="text-center">
                        <Link to='/signup' className="d-block small mt-3">Register an Account</Link>
                        <Link to='/signin' className="d-block small">Forgot Password?</Link>
                       </div>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        )
    }
}

export default SigninForm;