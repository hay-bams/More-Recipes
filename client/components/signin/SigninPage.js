import React from 'react';
import SiginForm from './SigninForm'; 
import Navigation from '../Navigation';

/**
 * @class SigninPage
 */
class SigninPage extends React.Component {
    render() {
      return (
        <div className='bg-dark'>
          <Navigation />
          <SiginForm />
        </div>
      )
    }
}

export default SigninPage;