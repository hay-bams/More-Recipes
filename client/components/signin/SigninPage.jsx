import React from 'react';
import PropTypes from 'prop-types';
import SiginForm from '../../containers/SigninForm';
import Navigation from '../Navigation';

const SigninPage = props => (
  <div>
    <Navigation  {...props} />
    <SiginForm redirectUser={props.history} />
  </div>
);

SigninPage.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func
  }).isRequired
};

export default SigninPage;
