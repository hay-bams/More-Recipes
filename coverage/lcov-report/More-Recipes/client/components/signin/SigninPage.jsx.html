<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for More-Recipes/client/components/signin/SigninPage.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">More-Recipes/client/components/signin</a> SigninPage.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import ConnectedSigninForm from '../../containers/SigninForm';
import { Navigation } from '../Navigation';
&nbsp;
&nbsp;
export const SigninPage = props =&gt; (
  &lt;div&gt;
    {
      localStorage.userData !== undefined ?  <span class="branch-0 cbranch-no" title="branch not covered" >props.history.push('/') </span>: ''
      }
    &lt;Navigation  {...props} /&gt;
    &lt;ConnectedSigninForm redirectUser={props.history} /&gt;
  &lt;/div&gt;
);
&nbsp;
SigninPage.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func
  }).isRequired
};
&nbsp;
export default SigninPage;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Mar 13 2018 14:50:33 GMT+0100 (WAT)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
