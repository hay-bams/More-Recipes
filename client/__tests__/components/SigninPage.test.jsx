import React from 'react';
import { shallow } from 'enzyme';
import { SigninPage } from '../../components/signin/SigninPage';

describe('Sign in page', () => {
  test('it should render correctly', () => {
    const props = {
      history: {
        push: jest.fn()
      }
    };

    const wrapper = shallow(<SigninPage {...props} />);
    expect(wrapper.find('Navigation').length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
});
 