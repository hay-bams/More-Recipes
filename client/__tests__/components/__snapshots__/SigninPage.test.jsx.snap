// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign in page it should render correctly 1`] = `
<div>
  <Navigation
    history={
      Object {
        "push": [MockFunction],
      }
    }
    location={
      Object {
        "pathname": "",
      }
    }
    userData={Object {}}
  />
  <Connect(SigninForm)
    redirectUser={
      Object {
        "push": [MockFunction],
      }
    }
  />
</div>
`;
